src/generic/stage2_build_tape.h

```diff
@@ -413,6 +414,7 @@ WARN_UNUSED error_code implementation::stage2(const uint8_t *buf, size_t len, pa
   }
 
 main_array_switch:
+  doc_parser.containing_scope_count[parser.depth-1] ++; // we have a new value
```

Comment:
You may be hitting the fiddly stage 2 blues here.

My memory is a little hazy, but this triggered me when I saw it: I recall trying to de-duplicate some code by moving it to the beginning of object_key_state / main_array_switch. You might try moving this so that it's in array_continue / array_begin and see if anything gets faster ...

Reply:
Yes.

Reply:
I moved them a bit around and it helped. Weird.